main/main.cpp

```diff
@@ -1666,6 +1671,8 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 			debug_uri = "local://";
 			OS::get_singleton()->_debug_stdout = true;
 #if defined(DEBUG_ENABLED)
+		} else if (arg == "--fail-on-error") {
+			fail_on_error = true;
```

Comment:
Remember to move this above the define if this is to be exposed in release builds.